pkg/apis/eksctl.io/v1alpha3/types.go

```diff
@@ -181,31 +241,48 @@ func (c *ClusterConfig) NewNodeGroup() *NodeGroup {
 // NodeGroup holds all configuration attributes that are
 // specific to a nodegroup
 type NodeGroup struct {
-	Name string
-
-	AMI               string
-	AMIFamily         string
-	InstanceType      string
-	AvailabilityZones []string
-	Tags              map[string]string
-	PrivateNetworking bool
-
-	DesiredCapacity int
-	MinSize         int
-	MaxSize         int
-
-	VolumeSize int
-
-	Labels         NodeLabels
-	MaxPodsPerNode int
-
-	PolicyARNs      []string
-	InstanceRoleARN string
-
-	AllowSSH         bool
-	SSHPublicKeyPath string
-	SSHPublicKey     []byte
-	SSHPublicKeyName string
+	Name string `json:"name"`
+	// +optional
+	AMI string `json:"ami,omitempty"`
+	// +optional
+	AMIFamily string `json:"amiFamily,omitempty"`
+	// +optional
+	InstanceType string `json:"instanceType,omitempty"`
+	// +optional
+	AvailabilityZones []string `json:"availabilityZones,omitempty"`
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+	// +optional
+	PrivateNetworking bool `json:"privateNetworking"`
+
+	DesiredCapacity int `json:"desiredCapacity"`
```

Comment:
I believe `DesiredCapacity` should be optional. When an external autoscaler that controls the capacity, like`cluster-autoscaler`, is enabled, you don't want to re-add/delete ec2 instances removed/added by the autoscaler.

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-as-group-desiredcapacity

Reply:
At the moment this is entirely open, and we have #385, please suggest there or feel free to open another issue to discuss.

Reply:
Opened #395 for this 👍